reV/config/project_points.py

```diff
@@ -676,6 +448,166 @@ def h(self):
 
         return self._h
 
+    @property
+    def curtailment(self):
+        """Get the curtailment config object.
+
+        Returns
+        -------
+        _curtailment : NoneType | reV.config.curtailment.Curtailment
+            None if no curtailment, reV curtailment config object if
+            curtailment is being assessed.
+        """
+        return self._curtailment
+
+    @staticmethod
+    def _parse_csv(fname):
+        """Import project points from .csv
+
+        Parameters
+        ----------
+        fname : str
+            Project points .csv file (with path). Must have 'gid' and 'config'
+            column names.
+
+        Returns
+        -------
+        df : pd.DataFrame
+            DataFrame mapping sites (gids) to SAM technology (config)
+        """
+        if fname.endswith('.csv'):
+            df = pd.read_csv(fname)
+        else:
+            raise ValueError('Config project points file must be '
+                             '.csv, but received: {}'
+                             .format(fname))
+
+        return df
+
+    @staticmethod
+    def _parse_sites(points, res_file=None):
+        """Parse project points from list or slice
+
+        Parameters
+        ----------
+        points : str | pd.DataFrame | slice | list
+            Slice specifying project points, string pointing to a project
+            points csv, or a dataframe containing the effective csv contents.
+        res_file : str | NoneType
+            Optional resource file to find maximum length of project points if
+            points slice stop is None.
+
+        Returns
+        -------
+        df : pd.DataFrame
+            DataFrame mapping sites (gids) to SAM technology (config)
+        """
+        df = pd.DataFrame(columns=['gid', 'config'])
+        if isinstance(points, (list, tuple)):
+            # explicit site list, set directly
+            df['gid'] = points
+        elif isinstance(points, slice):
+            stop = points.stop
+            if stop is None:
+                if res_file is None:
+                    raise ValueError('Must supply a resource file if '
+                                     'points is a slice of type '
+                                     ' slice(*, None, *)')
+
+                multi_h5_res, _ = check_res_file(res_file)
+                if multi_h5_res:
+                    stop = MultiFileResource(res_file).shape[1]
+                else:
+                    stop = Resource(res_file).shape[1]
+
+            df['gid'] = list(range(*points.indices(stop)))
+        else:
+            raise TypeError('Project Points sites needs to be set as a list, '
+                            'tuple, or slice, but was set as: {}'
+                            .format(type(points)))
+
+        df['config'] = None
+
+        return df
+
+    @classmethod
+    def _parse_points(cls, points, res_file=None):
```

Comment:
Whats the motivation for moving all these methods around? I'm all for re-org but it makes reviewing the diff's impractical.

Reply:
Yeah sorry on the diff issues, It was re-org for readability